### DISCUSSION TITLE
Child Object Validation Rules and Blazor

### DISCUSSION THREAD

**Discussion by bashinamwila** [MarimerLLC/csla#2087] [Questions] [answered]
I have BO which has child object which is also a subclass of BusinessBase.The parent object has two properties which have simple DataAnnotation Required Rule and the Child Object also has one property which also has simple data annotation rule.When the properties  are bound to TextInput component as follows
`<TextInput Property=vm.GetPropertyInfo(()=>vm.Model.Property1) />`
`<TextInput Property=vm.GetPropertyInfo(()=>vm.Model.Property2) />`
`<TextInput Property=vm.GetPropertyInfo(()=>vm.Model.ChildObject.Property1) />`
The first two TextInput's show the ErrorText, whilst the TextInput bound to child object does not show the any validation error,
and so even when all properties have text values the Save button still remain disabled.
Is anyone aware what is causing this behavoiur,Your help deeply appreciated.
Many Thanks
bashinamwila

**Comment by bashinamwila** [comment, marked as answer]
I found something on how validate complex types like scenario above using ObjectGraphDataAnnotationsValidator and the ValidateComplexType attribute from the Microsoft.AspNetCore.Components.DataAnnotations.Validations nuget for the EditForm component,I am using the ViewModel looking for help on how I can achieve this with ViewModel in Csla.Blazor

**Reply by rockfordlhotka** [reply]
I am not sure I understand the problem.

Given this parent-child graph:

```c#
using System;
using System.ComponentModel.DataAnnotations;
using Csla;

namespace BlazorApp3.Data
{
  [Serializable]
  public class PersonEdit : BusinessBase<PersonEdit>
  {
    public static readonly PropertyInfo<string> NameProperty = RegisterProperty<string>(nameof(Name));
    [Required]
    public string Name
    {
      get => GetProperty(NameProperty);
      set => SetProperty(NameProperty, value);
    }

    public static readonly PropertyInfo<PhoneNumberEdit> PhoneProperty = RegisterProperty<PhoneNumberEdit>(nameof(Phone));
    public PhoneNumberEdit Phone
    {
      get => GetProperty(PhoneProperty);
      private set => LoadProperty(PhoneProperty, value);
    }

    [Fetch]
    private void Fetch(string name)
    {
      using (BypassPropertyChecks)
      {
        Name = name;
        Phone = DataPortal.FetchChild<PhoneNumberEdit>("555-123-4567");
      }
    }
  }

  [Serializable]
  public class PhoneNumberEdit : BusinessBase<PhoneNumberEdit>
  {
    public static readonly PropertyInfo<string> PhoneNumberProperty = RegisterProperty<string>(nameof(PhoneNumber));
    [Required]
    public string PhoneNumber
    {
      get => GetProperty(PhoneNumberProperty);
      set => SetProperty(PhoneNumberProperty, value);
    }

    [FetchChild]
    private void Fetch(string number)
    {
      using (BypassPropertyChecks)
      {
        PhoneNumber = number;
      }
    }
  }
}
```

And this page code:

```
@page "/"
@inject Csla.Blazor.ViewModel<Data.PersonEdit> vm

@if (vm.Model != null)
{
<div><TextInput Property="@(vm.GetPropertyInfo(() => vm.Model.Name))" /></div>
<div><TextInput Property="@(vm.GetPropertyInfo(() => vm.Model.Phone.PhoneNumber))" /></div>
}

@code {
  protected override async Task OnParametersSetAsync()
  {
    await vm.RefreshAsync(() => Csla.DataPortal.FetchAsync<Data.PersonEdit>("Juan"));
  }
}
```

If I blank both fields I see this:

![image](https://user-images.githubusercontent.com/2333134/106972761-22c24d80-6717-11eb-9925-6d600f250deb.png)

**Reply by rockfordlhotka** [reply]
This is using the `TextInput` component from the `ProjectTracker` sample.

**Reply by bashinamwila** [reply]
Thank you @rockfordlhotka what I forgot to include in the dataportal create was a call for the BusinessRules.CheckRules.Thank you very much

**Reply by jhuerta3** [reply]
Thank you @rockfordlhotka I had the same issue.
